django/contrib/contenttypes/fields.py

```diff
@@ -250,7 +250,7 @@ def __set__(self, instance, value):
         if value is not None:
             ct = self.get_content_type(obj=value)
             fk = value._get_pk_val()
-            if fk is None:
+            if not getattr(self, 'allow_unsaved_instance_assignment', False) and fk is None:
```

Comment:
is there a reason to use `getattr` instead of defining the attribute as `False` on the class?